### ISSUE TITLE
ES6Promise reference and missing Promise reject handling

### ISSUE THREAD

**Issue by orendin** [PredixDev/px-login#14]
Hi there

I have started developing an Angular2 (rc6) app and am using px-login. While working on this, I noticed that my Angular page did not do proper change detection, failing to update the view when a property changed. After a day of puzzling I finally found out that the es6-promise library is interfering with Angular's zones and promise handling. In Angular2, they are now suggesting to use core-js, which already includes a es6 promise shim. 
Somehow core-js and es-6 promise bite each other in a way which makes Angular2 fail to update the view silently.

The next step was to remove the es-6 promise reference which I included for px-login to work. This fails since you do a conditional loading on a hard-coded, relative folder name.

```
//Conditional loading of es6-promise polyfill
    if(typeof ES6Promise === "undefined"){
      var script = document.createElement('script');
      script.async = true;
      script.src = '../es6-promise/promise.min.js';
      document.head.appendChild(script);
    }

```

In my opinion, this is really bad! You can't just assume that some js file is there in a relative path from inside a Polymer component. Show an error message, log a console message, but don't try and load it magically. It has to be the page / apps responsibility to load a library that implements ES6Promises.

Fundamentally, after commenting this out, my app still works. But I get another error from the getUserInfo function when it tries to get the user info:

> Unhandled Promise rejection: Unexpected token < in JSON at position 0 ; Zone: <root> ; Task: Promise.then ; Value: SyntaxError: Unexpected token < in JSON at position 0

getUserInfo fails and rejects the Promise properly, but the error is not handled in the ready function:

```
 window.px.auth.getUserInfo().then(function (data) {
                  self.userInfo = data;
                });
```

This needs a catch like so:

```
.catch(function(){
    //do something with the error.
})
```

**Comment by gedesigntech** [comment]
hi @orendin thanks for the fix to this particular issue of px-login. Sorry for the trouble you had to go through.  We have been aware of this and other items, and have them in our list for px-login.  We are looking into addressing all of them in a possibly major re-work of the component.  Your solution clearly shows best practice, which the next release of the component will adhere to. Please stay tuned for updates.  Thanks again!

**Comment by randyaskin** [comment]
I believe this can be closed per the 1.0+ redesign as Henry indicated above.
